### ISSUE TITLE
get_parameter/artifact for template invocators (Steps/DAG)

### ISSUE THREAD

**Issue by elliotgunton** [argoproj-labs/hera#816]
**Is your feature request related to a problem? Please describe.**
Passing Steps/DAG inputs directly through to substeps/Tasks requires the expression syntax string or `expr`

**Describe the solution you'd like**
DAGs/Steps to expose `get_parameter`/`get_artifact` for use in child nodes

Currently, for an example, see [callable_dag.py](https://github.com/argoproj-labs/hera/blob/b13811ab8548bd4a169c330c8dd777ce0794e90e/examples/workflows/callable_dag.py#L13-L15
), but instead passing the value through directly:

```py
    with DAG(name="my-dag", inputs=Parameter(name="my-dag-input")) as my_dag:
        hello(name="hello", arguments={"name": "{{inputs.parameters.my-dag-input}}"})
```

Preferred would be:

```py
    with DAG(name="my-dag", inputs=Parameter(name="my-dag-input")) as my_dag:
        hello(name="hello", arguments=[my_dag.get_parameter("my-dag-input").with_name("name")])
```

**Describe alternatives you've considered**
This will help us get away from requiring expr knowledge, but can be used for now.

**Additional context**

We have some work in https://github.com/argoproj-labs/hera/pull/778 for output references, should think about how these can work together, also the callable syntax in #763

**Comment by vikramsg** [comment]
Just came across this issue today. Would be very keen on using this. Let me know if I could help.

**Comment by elliotgunton** [comment]
`get_artifact` was never implemented - added #1283
